extensions/amp-analytics/0.1/vendors/snowplow_v2.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {jsonLiteral} from '../../../../src/json';
+
+const SNOWPLOW_V2_CONFIG = jsonLiteral({
+  'linkers': {
+    'linker': {
+      'ids': {
+        '_sp_duid': '${duid}',
+      },
+    },
+  },
+  'cookies': {
+    'enabled': true,
+    'cookieMaxAge': 1800,
+    '_sp_duid': {
+      'value':
+        '$IF($IF(COOKIE(_sp_duid), COOKIE(_sp_duid), LINKER_PARAM(linker,_sp_duid)), $IF(COOKIE(_sp_duid), COOKIE(_sp_duid), LINKER_PARAM(linker,_sp_duid)), $SUBSTR(QUERY_PARAM(_sp),0,36))',
+    },
+  },
+  'vars': {
+    'aaVersion': 'amp-1.0.0',
+    'ampVistorId': 'CLIENT_ID(_sp_id)',
+    'nullString': 'null',
+    'customEventTemplate': [
+      '{',
+      '"schema":"iglu:com.snowplowanalytics.snowplow/unstruct_event/jsonschema/1-0-0",',
+      '"data":{',
+      '"schema":"iglu:${customEventSchemaVendor}/${customEventSchemaName}/jsonschema/${customEventSchemaVersion}",',
+      '"data":{',
+      '${customEventSchemaData}',
+      '}',
+      '}',
+      '}',
+    ].join(''),
+    'contextsHead':
+      '{"schema":"iglu:com.snowplowanalytics.snowplow/contexts/jsonschema/1-0-0","data":[{"schema":"iglu:com.snowplowanalytics.amp/amp_id/jsonschema/1-0-0","data":{"ampClientId":"${ampVistorId}", "domainUserId": "COOKIE(_sp_duid)", "userId": "${userId}"}},${customContexts}',
+    'contextsTail':
+      '{"schema":"iglu:com.snowplowanalytics.amp/amp_web_page/jsonschema/1-0-0","data":{"ampPageViewId":"PAGE_VIEW_ID_64"}}]}',
```

Comment:
Any scope for more properties here? Particularly [`VIEWER`](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md#viewer), and one or both of [`SOURCE_URL`](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md#source-url) and [`CANONICAL_URL`](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md#canonical-url) would be useful, especially for publishers who would be keen to aggregate metrics by article/URL which will now be different given it reflects `ampdocUrl`. Or is this just left open for people to send through in a custom context?

Reply:
I think for now these will have to be left to a custom context - both because we should keep the scope limited for now, and because depending on the length of those urls, we're eating up the character limit for a GET request.

Reply:
I've added this as [an issue](https://github.com/snowplow-incubator/amphtml/issues/2) for futture consideration.